tests/unit/test_suite_config_upgrader.py

```diff
@@ -0,0 +1,212 @@
+    the config passes through the function unaltered.
+    """
+    assert host_to_platform_upgrader(get_task_conf) == get_task_conf
+
+
+def test_platform_and_harmless_old_items__doesnothing(get_task_conf):
+    """Double check that when old items such as [job][execution time limit]
+    are added to the config the config is returned unchanged.
+    """
+    config = get_task_conf
+    config['runtime']['sometask'].update({'execution time limt': 42})
+    assert host_to_platform_upgrader(config) == \
+        get_task_conf
+
+
+def test_noplatform_no_old_items(get_task_conf):
+    conf = host_to_platform_upgrader(get_task_conf)
+    assert conf['runtime']['sometask']['platform'] == 'localhost'
+
+
+def test_platform_and_default_old_item_return_platform_localhost(
+    get_task_conf
+):
+    """In the suite config "[runtime][remote]batch system" has a default
+    value, which should not be the cause of a failure to upgrade.
+    """
+    conf = get_task_conf
+    conf['runtime']['sometask'].update({
+        'platform': 'localhost',
+        'job': {'batch system': ''}
+    })
+    conf = host_to_platform_upgrader(conf)
+    assert conf['runtime']['sometask']['platform'] == 'localhost'
+
+
+def test_noplatform_hostfunction__logsdebug(caplog, get_task_conf):
+    caplog.set_level(logging.DEBUG)
+    config = get_task_conf
+    config['runtime']['sometask'].update(
+        {'remote': {'host': '$(echo hi)'}}
+    )
+    assert host_to_platform_upgrader(config) == config
+    assert "'sometask' is a function" in caplog.messages[0]
+
+
+def test_noplatform_hostname__selectplatform(
+    get_task_conf, mock_glbl_cfg, caplog
+):
+    """
+    Check that we can select a platform and remove forbidden items
+
+    This test does not need to replicate the tests for the
+    `platform_from_job_info` (a.k.a Reverse Lookup), just confirm
+    that we are using that function correctly here.
+    """
+    caplog.set_level(logging.WARNING)
+    mock_glbl_cfg(
+        'cylc.flow.cfgspec.suite.glbl_cfg',
+        '''
+        [platforms]
+            [[saffron]]
+                hosts = saff01
+                batch system = slurm
+        '''
+    )
+    config = get_task_conf
+    config['runtime']['sometask'].update(
+        {
+            'remote': {'host': 'saff01'},
+            'job': {'batch system': 'slurm'}
+        }
+    )
+    assert host_to_platform_upgrader(config) == \
+        {'runtime': {'sometask': {
+            'remote': {},
+            'job': {},
+            'platform': 'saffron'
+        }}}
+    assert "Cylc 8 platform \"saffron\" selected" in caplog.messages[0]
+
+
+def test_noplatform_hostname__cantselectplatform(
+    get_task_conf, mock_glbl_cfg
+):
+    """
+    Check that we can select a platform and remove forbidden items
+
+    This test does not need to replicate the tests for the
+    `platform_from_job_info` (a.k.a Reverse Lookup), just confirm
+    that we are using that function correctly here.
+    """
```

Comment:
Docstring copied from test above - needs updating to better describe this test

Reply:
🐆